perun-openapi/openapi.yml

```diff
@@ -8290,296 +8354,1032 @@ paths:
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfRichDestinationsResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/servicesManager/getAllRichDestinations/f:
+    get:
+      tags:
+        - ServicesManager
+      operationId: getAllRichDestinationsForFacility
+      summary: Returns list of all rich destinations defined for the facility.
+      parameters:
+        - $ref: '#/components/parameters/facilityId'
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfRichDestinationsResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/servicesManager/getRichDestinations:
+    get:
+      tags:
+        - ServicesManager
+      operationId: getRichDestinations
+      summary: Returns list of all rich destinations defined for the service and facility.
+      parameters:
+        - $ref: '#/components/parameters/serviceId'
+        - $ref: '#/components/parameters/facilityId'
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfRichDestinationsResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /urlinjsonout/servicesManager/addDestination:
+    post:
+      tags:
+        - ServicesManager
+      operationId: addDestination
+      summary: Adds an destination for a facility and service. If destination doesn't exist it will be created.
+      parameters:
+        - $ref: '#/components/parameters/serviceId'
+        - $ref: '#/components/parameters/facilityId'
+        - { name: destination, description: "Destination's destination, usually a DNS name", schema: { type: string },  in: query, required: true }
+        - { name: type, description: "Destination's type", schema: { $ref: '#/components/schemas/DestinationType' },  in: query, required: true }
+        - { name: propagationType, description: "Destination's propagationType", schema: { $ref: '#/components/schemas/DestinationPropagationType' },  in: query, required: true }
+      responses:
+        '200':
+          $ref: '#/components/responses/DestinationResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/servicesManager/addDestination:
+    post:
+      tags:
+        - ServicesManager
+      operationId: addDestinationToMultipleServices
+      description: Adds an destination for a facility and list of services. If destination doesn't exist it will be created.
+      responses:
+        '200':
+          $ref: '#/components/responses/DestinationResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              title: InputAddDestinationToMultipleServices
+              description: "input to addDestinationToMultipleServices"
+              type: object
+              required:
```

Comment:
If all of the possible input fields are required, it is impossible to call this method with only some of them, which is actually allowed by the implementation. You should probably split this method into multiple versions.

Reply:
@xkureck this is not fixed